src/common/Spine.ts

```diff
@@ -139,8 +144,15 @@ export class Spine<N extends IR.Node = IR.Node> {
     skipReplaced = false,
     skipThis = false,
   ): Spine {
-    const ret = skipThis ? undefined : replacer(this.node, this);
-    if (ret === undefined) {
+    let skipReplacement = skipReplaced;
+    let skipChildren = false;
+    function skip(kind: "replacement" | "children") {
+      skipReplacement ||= kind === "replacement";
+      skipChildren ||= kind === "children";
```

Comment:
As I understand it:
- `skipReplacement` prevents recursion into the new node (if there is a replacement)
- `skipChildren` prevents recursion into the new node (if there is a replacement) *and* into the children of the old node.
But I'm not sure about this behavior. Could we have a comment clarifying? A good place for the comment would be on the `VisitorContext` interface, if you go with that idea:
```
interface VisitorContext {
  skipReplacement: () => void;
  skipChildren: () => void;
}
```

Reply:
Yeah sorry, it's confusing. I went trough several iterations of what it is I actually want it to do.